third_party/move/move-compiler-v2/tests/bytecode-generator/wildcard3.exp

```diff
@@ -0,0 +1,56 @@
+// -- Model dump before bytecode pipeline
+module 0xc0ffee::m {
+    struct S {
+        dummy_field: bool,
+    }
+    public fun bar() {
+        {
+          let s: m::S = pack m::S(false);
+          {
+            let _: m::S = s;
+            Tuple()
+          }
+        }
+    }
+    public fun foo(s: m::S) {
+        {
+          let _: m::S = s;
+          Tuple()
+        }
+    }
+} // end 0xc0ffee::m
+
+============ initial bytecode ================
+
+[variant baseline]
+public fun m::bar() {
+     var $t0: m::S
+     var $t1: bool
+     var $t2: m::S
+  0: $t1 := false
+  1: $t0 := pack m::S($t1)
+  2: $t2 := infer($t0)
+  3: return ()
+}
+
+
+[variant baseline]
+public fun m::foo($t0: m::S) {
+     var $t1: m::S
+  0: $t1 := infer($t0)
+  1: return ()
+}
+
+
+Diagnostics:
+error: value of type `m::S` does not have the `drop` ability
```

Comment:
I notice that V1 shows the reason for the error type again:
```
error[E05001]: ability constraint not satisfied
  ┌─ tests/move_check/v2/wildcard3.move:5:13
  │
2 │     struct S {}
  │            - To satisfy the constraint, the 'drop' ability would need to be added here
3 │ 
4 │     public fun foo(s: S) {
  │                       - The type '0xC0FFEE::m::S' does not have the ability 'drop'
5 │         let _ = s;
  │             ^ Cannot ignore values without the 'drop' ability. The value must be used
```
Please add this test case to existing bug (or file one) to improve the error message for such cases.

Reply:
I find what we are suggesting here pretty reasonable. We also show the reason (implicitly dropped here ...).

This also has nothing to do with this PR (exact same error is reported with non-wildcard variable).

Due to the above reasons, I am not filing a bug. But if you feel strongly about some aspect of this reporting, do feel free to file one.

Reply:
Well, in this case there is a declared type.  It's the inferred types that can be nonobvious.